crates/pathfinder/src/bin/pathfinder.rs

```diff
@@ -29,10 +31,16 @@ async fn main() -> anyhow::Result<()> {
 
     permission_check(&config.data_directory)?;
 
+    let prometheus_handle = PrometheusBuilder::new()
+        .install_recorder()
+        .context("Creating Prometheus recorder")?;
+
     let pathfinder_ready = match config.monitoring_addr {
         Some(monitoring_addr) => {
             let ready = Arc::new(AtomicBool::new(false));
-            let _jh = monitoring::spawn_server(monitoring_addr, ready.clone()).await;
+            let prometheus_handle = prometheus_handle;
+            let _jh =
+                monitoring::spawn_server(monitoring_addr, ready.clone(), prometheus_handle).await;
```

Comment:
We may want to consider a separate config enabling metrics; should ask API services if they would want these.